lib/tar.ml

```diff
@@ -815,3 +815,44 @@ let encode_header ?level header =
 
 let encode_global_extended_header ?level global =
   encode_extended_header ?level `Global global
+
+type ('a, 'err) t =
+  | Really_read : int -> (string, 'err) t
+  | Read : int -> (string, 'err) t
+  | Seek : int -> (int, 'err) t
+  | Bind : ('a, 'err) t * ('a -> ('b, 'err) t) -> ('b, 'err) t
```

Comment:
For `Bind (x, f)` the values `x` and `f` share the same `'err`. This is likely just fine, but it could maybe be generalized (though I don't think it's worth it going through those hoops).